src/main/java/net/masterthought/cucumber/generators/AbstractPage.java

```diff
@@ -111,6 +111,16 @@ private void buildGeneralParameters() {
         context.put("build_project_name", configuration.getProjectName());
         context.put("build_number", configuration.getBuildNumber());
 
+        String buildUrl = configuration.getBuildUrl();
+        if (StringUtils.isNotBlank(buildUrl)) {
+            context.put("build_url", buildUrl);
+            String buildName = configuration.getBuildName();
+            if (StringUtils.isBlank(buildName)) {
+                buildName = "Build #" + configuration.getBuildNumber();
+            }
+            context.put("build_name", buildName);
```

Comment:
... and build name at the same time ?

Reply:
The build number has attached semantics in your case, meaning it must be an integer that is consecutive, so you can determine any previous build link / result link from it in the Jenkins use case.
My use case is however entirely different. First I use Gitlab that has non-consecutive build IDs, secondly the reports cannot be attached to an existing Job / Pipeline directly, but have to be put somewhere else, meaning that a build ID alone without the context of a Job or Pipeline is meaningless.

So, no, both at the same time make no sense, a build number makes most sense in a Jenkins Job environment, while the more generic build name / URL thing I introduced is better for a non-structured build / reporting environment.

If you say you want to get rid / replace the build number by this, fine, I can remove this part, but I did not want to do that before talking to you :)